apstools/callbacks/callback_base.py

```diff
@@ -60,16 +61,16 @@ class FileWriterCallbackBase:
        ~stop
     """
 
-    file_extension = "dat"
-    _file_name = None
-    _file_path = None
+    file_extension: str = "dat"
+    _file_name: Optional[pathlib.Path] = None
```

Comment:
For both of these, is `Optional[Union[pathlib.Path,str]]` acceptable?  I think so.  Check the unit testing to see if both ways are tested.

Reply:
Updated. Checking the tests soon.

Reply:
Created a new test. Need to check if is redundant with others.